Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using AzureCognitiveSearch.PowerSkills.Common;
+using Newtonsoft.Json.Linq;
+
+namespace AzureCognitiveSearch.PowerSkills.Text.CustomEntitySearch
+{
+    /// <summary>
+    /// Based on sample custom skill provided in Azure Search. Provided a user-defined list of entities
+    /// this function determines the entities first occurrence within a given document. This list of entities
+    /// must repeatedly be provided by the user for each document.
+    /// </summary>
+    public static class CustomEntitySearch
+    {
+        /// <summary>
+        /// We assert the following assumptions:
+        /// 1. All text files are Latin based (i.e. no languageCode)
+        /// 2. Words can contain special characters and numbers
+        /// 3. The provided entities are not case sensitive
+        /// </summary>
+
+        [FunctionName("custom-search")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log, ExecutionContext executionContext)
+        {
+            log.LogInformation("Custom Entity Search function: C# HTTP trigger function processed a request.");
+
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            const int MAXTIME = 1;
```

Comment:
>MAXTIME [](start = 22, length = 7)

nit: use descriptive variable names. EG: MaxRegexEvaluationTime

Reply:
nit: can also move this to a static class const rather than defining in function body

---
In reply to: [304663420](https://github.com/Azure-Samples/azure-search-power-skills/pull/1#discussion_r304663420) [](ancestors = 304663420)